astroid/nodes/node_classes.py

```diff
@@ -1062,7 +1062,12 @@ def _infer_attribute(
             if isinstance(owner, (ClassDef, Instance)):
                 frame = owner if isinstance(owner, ClassDef) else owner._proxied
                 context.constraints[node.attrname] = get_constraints(node, frame=frame)
-            yield from owner.igetattr(node.attrname, context)
+            if node.attrname == "argv" and owner.name == "sys":
```

Comment:
I'm not sure how often this happens, but this could conflict with implementations of `sys`. Perhaps we should just add a `sys` brain to handle determining whether it is `buitlin.sys`?

Reply:
That's a false positive in pylint, and it happens relatively often (some duplicate opened for it) see https://github.com/canonical/apport/commit/b4e81f4d196a148259f2429aaae7a68f84d6dd24 for example. Maybe the brain approach is less costly ? But in general I think performance are less important than accuracy for a linter that is already slow.

Reply:
Yeah I think we need a brain and use inference here.

Reply:
How would you test the performance diff ?

Reply:
@DanielNoord why do you think this is worth the inference penalty? You're imagining a custom script called "sys" that would lose the ability to infer an "argv"? I'm willing to let that be unhandled.

Reply:
Mostly because you can't control it as a user. If we want to prefer false negatives over false positives I think it is not nice that the user can't control this.

But if you guys think this is fine then go ahead!

Reply:
I think ``sys.argv`` is actually ``Uninferable``. Often  ``Uninferable`` is 'soft' uninferable as in "our parsing does not permit to realize the variable MUST be none at this point because we do not have control flow so let's say we don't know", but I think this is an example of 'hard' ``Uninferable`` as in "we'll never know for sure what's in it, during static analysis".  How would the user control what is in ``sys.argv`` through a brain ? It seems to me that they face the same problem than us. If we want to give control to users over false negative vs false positive on ``Uninferable`` values, changes needs to be happening directly in pylint imo (but then every single astroid deficiency would create tons of FP).

Reply:
I agree that `stdlib.sys.argv` is `Uninferable`, but `user_module.sys.argv` isn't necessarily. That's why we would normally use inference: to see whether `sys` is builtin or from some user defined code.

Reply:
Ha, right, so we should use qname then ? Although I'm not sure how often someone would happen to mimick sys.argv while also changing the meaning of sys.argv by putting something other than runtime args in it. (Or if this person deserves a linter 😛 )

Reply:
I'm glad it was discussed, but yeah, I think we should stay focused on higher value cases and merge as is.

Reply:
Decided to just test this out. I don't think we have a real problem. I added a "sys" module to astroid, and with this PR:
```py
>>> from astroid import extract_node
>>> n1 = extract_node("import astroid.sys; astroid.sys.argv")
>>> n1.inferred()[0]
<Const.int l.1 at 0x104c83f50>
>>> n2 = extract_node("import sys; sys.argv")
>>> n2.inferred()[0]
Uninferable
>>> n3 = extract_node("import astroid.sys as sys; sys.argv")
>>> n3.inferred()[0]
<Const.int l.1 at 0x104c83f50>
```